tests/test_builders/test_build_linkcheck.py

```diff
@@ -99,9 +102,44 @@ def connection_count(self):
         return len(self.connections)
 
 
+@contextmanager
+def rewrite_hyperlinks(app, server):
+    """
+    Rewrite hyperlinks that refer to network location 'localhost:7777',
+    allowing that location to vary dynamically with the arbitrary test HTTP
+    server port assigned during unit testing.
+
+    :param app: The Sphinx application where link replacement is to occur.
+    :param server: Destination server to redirect the hyperlinks to.
+    """
+    match_netloc, replacement_netloc = (
+        'localhost:7777',
+        f'localhost:{server.server_port}',
+    )
+
+    def rewrite_hyperlink(_app, uri: str) -> str | None:
+        parsed_uri = urlparse(uri)
+        if parsed_uri.netloc != match_netloc:
+            return uri
+        return parsed_uri._replace(netloc=replacement_netloc).geturl()
+
+    listener_id = app.connect('linkcheck-process-uri', rewrite_hyperlink)
+    yield
+    app.disconnect(listener_id)
+
+
+@contextmanager
+def serve_html(app, handler, *, tls_enabled=False):
+    with (
+        http_server(handler, tls_enabled=tls_enabled) as server,
+        rewrite_hyperlinks(app, server),
+    ):
+        yield f'localhost:{server.server_port}'
+
+
 @pytest.mark.sphinx('linkcheck', testroot='linkcheck', freshenv=True)
 def test_defaults(app):
-    with http_server(DefaultsHandler):
+    with serve_html(app, DefaultsHandler) as test_server:
```

Comment:
```suggestion
    with serve_html(app, DefaultsHandler) as address:
```

and change the references after

Reply:
In theory I agree that `address` is a more logical name here.  `{test_server}` has the benefit that it has the same string length as `localhost`, making the diff marginally easier to inspect.

I'll probably change it anyway.. but just a note that this is something I'm considering.

Reply:
Honestly, the diff issue has less priority than what the variable should explain. `test_server` could mean many things in a test file and it would mean, in the first case, a server object rather than an address IMO.